CRM/Report/Form/Contact/Detail.php

```diff
@@ -608,15 +606,18 @@ public function where() {
    */
   public function clauseComponent() {
     $selectedContacts = implode(',', $this->_contactSelected);
-    $contribution = $membership = $participant = NULL;
     $eligibleResult = $rows = $tempArray = array();
     foreach ($this->_component as $val) {
       if (!empty($this->_selectComponent[$val]) &&
         ($val != 'activity_civireport' && $val != 'relationship_civireport')
       ) {
+        $tableName = str_replace('_civireport', '', $val);
```

Comment:
I am just trying to get the table alias defined for table from $this->_component since normally components are defined as {table_name}_civireport, So i get table name from $this->_component and use later to get alias defined for it. I am using the pattern normally followed in report.